pageserver/src/tenant/timeline.rs

```diff
@@ -669,7 +736,7 @@ impl Timeline {
         // Is the timeline being deleted?
         let state = *self.state.borrow();
         if state == TimelineState::Stopping {
-            anyhow::bail!("timeline is Stopping");
+            return Err(anyhow::anyhow!("timeline is Stopping").into());
```

Comment:
shouldn't the lock be taken after this check ?

Reply:
The lock guards against concurrent layer removals (which for example eviction does) ... I don't think the order here matters; for this checked state the state change does not take the same lock.

I guess this does seem a bit out of place check for me, but it's not wrong.

Reply:
Chatted, this was a generic point about preconditions before taking any locks. I am inclined not to change this on this PR, current change is only for the introduced error type. Resolving.